### PR TITLE
feat(frontend): onSend callbacks in btc-send.services

### PR DESCRIPTION
**Pull request by DenysKarmazynDFINITY**
# Motivation

The idea is to decouple btc-send.services from the send flow so the service can be re-used by the conversion wizard too.

### CODE REVIEW

File: src/frontend/src/btc/services/btc-send.services.ts
Comment (AntonioVentilii) [inline src/frontend/src/btc/services/btc-send.services.ts:82]: should you omit `onSendCompleted`? or is there a future use-case?

**Comment by DenysKarmazynDFINITY** [comment]
@peterpeterparker @AntonioVentilii please take a look at https://github.com/dfinity/oisy-wallet/pull/3553 first. I'll come back to this PR after the tests are merged.

**Review by peterpeterparker — COMMENTED** [review]
I have to say I rather like a single callback. By having two, there is no guarantee that somehow someday, start and complete for some bug reason are triggered at the same time, you know what I mean?

**Comment by DenysKarmazynDFINITY** [comment]
@peterpeterparker yep, agree. 

@peterpeterparker @AntonioVentilii I have updated the code to use a single callback. WDYT?

**Review by peterpeterparker — APPROVED** [review]
I like `onProgress`, thanks.

Since @AntonioVentilii also participated maybe wait for his feedback too before merge.

File: src/frontend/src/btc/components/send/BtcSendTokenWizard.svelte
Comment (AntonioVentilii) [inline src/frontend/src/btc/components/send/BtcSendTokenWizard.svelte:119]: just a smaller question, unrelated: did the linter flagged something here or not?

**Review by AntonioVentilii — APPROVED** [review]

**Review by AntonioVentilii — APPROVED** [review]
LGTM! Tks

File: src/frontend/src/btc/components/send/BtcSendTokenWizard.svelte
Comment (DenysKarmazynDFINITY) [reply src/frontend/src/btc/components/send/BtcSendTokenWizard.svelte:119]: Nope, linter didn't complain here. I think it would do it if I had something like:
```
if (sendProgressStep === ProgressStepsSendBtc.INITIALIZATION) {
  progress(ProgressStepsSendBtc.SEND);
  return;
} else if (sendProgressStep === ProgressStepsSendBtc.SEND) {
...
```

**Pull request merged**
